server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoDataserviceService.java

```diff
@@ -0,0 +1,1733 @@
+                        dataServices[ i++ ] = this.wsMgr.resolve( uow, new ObjectImpl( this.wsMgr.getRepository(), path, 0 ), Dataservice.class );
+                    }
+
+                    LOGGER.debug( "getDataservices:found '{0}' DataServices using pattern '{1}'", dataServices.length, searchPattern ); //$NON-NLS-1$
+                }
+            }
+
+            int start = 0;
+
+            { // start query parameter
+                final String qparam = uriInfo.getQueryParameters().getFirst( QueryParamKeys.START );
+
+                if ( qparam != null ) {
+
+                    try {
+                        start = Integer.parseInt( qparam );
+
+                        if ( start < 0 ) {
+                            start = 0;
+                        }
+                    } catch ( final Exception e ) {
+                        start = 0;
+                    }
+                }
+            }
+
+            int size = ALL_AVAILABLE;
+
+            { // size query parameter
+                final String qparam = uriInfo.getQueryParameters().getFirst( QueryParamKeys.SIZE );
+
+                if ( qparam != null ) {
+
+                    try {
+                        size = Integer.parseInt( qparam );
+
+                        if ( size <= 0 ) {
+                            size = ALL_AVAILABLE;
+                        }
+                    } catch ( final Exception e ) {
+                        size = ALL_AVAILABLE;
+                    }
+                }
+            }
+
+            final List< RestDataservice > entities = new ArrayList< >();
+            int i = 0;
+
+            KomodoProperties properties = new KomodoProperties();
+            //properties.addProperty(VDB_EXPORT_XML_PROPERTY, false);
+            for ( final Dataservice dataService : dataServices ) {
+                if ( ( start == 0 ) || ( i >= start ) ) {
+                    if ( ( size == ALL_AVAILABLE ) || ( entities.size() < size ) ) {
+                        RestDataservice entity = entityFactory.create(dataService, uriInfo.getBaseUri(), uow, properties);
+                        entities.add(entity);
+                        LOGGER.debug("getDataservices:Dataservice '{0}' entity was constructed", dataService.getName(uow)); //$NON-NLS-1$
+                    } else {
+                        break;
+                    }
+                }
+
+                ++i;
+            }
+
+            // create response
+            return commit( uow, mediaTypes, entities );
+
+        } catch ( final Exception e ) {
+            if ( ( uow != null ) && ( uow.getState() != State.ROLLED_BACK ) ) {
+                uow.rollback();
+            }
+
+            if ( e instanceof KomodoRestException ) {
+                throw ( KomodoRestException )e;
+            }
+
+            return createErrorResponse(mediaTypes, e, DATASERVICE_SERVICE_GET_DATASERVICES_ERROR);
+        }
+    }
+    
+//    /**
```

Comment:
Lots of commented-out code in this class. Is it strictly necessary to be included?